### PR TITLE
Merge Cafebokning into master

### PR DESCRIPTION
**Pull request by Zofz**
Getting ready to open up the module for working in Hilbert Café, still have some refactor to do, do not merge it before I'm done.

Comments and ideas are great to hear! :D

### CODE REVIEW

**Comment by henrikssn** [comment]
- Alla tips av hound borde fixas.
- Commita inte saker från asset pipeline. (public/assets)
- Tester :)

**Comment by Zofz** [comment]
@henrikssn Yes, just wanted a view for the diff's. Not ready yet 😄 
public/assets are excluded in .gitgnore, or what?
Yep, want to have time for that as well 😄

**Comment by henrikssn** [comment]
@Zofz Yeah, see https://github.com/fsek/web/blob/master/.gitignore#L11

Ping me when you are ready for review :+1:

**Comment by Zofz** [comment]
@henrikssn Where did my public/assets get included? (They are removed when .gitignore is changed?).
Will do, perhaps some hours left over tonight. Time for exams...

On tests, I'm moving more "business logic" to the model and usually think that those tests are most important, atleast they are the ones that helps me most ;) 
You have done some controller-tests as well right?

**Comment by henrikssn** [comment]
@Zofz No idea, just remove them :)

Yeah, your controllers are very complex as it stands right now. Moving code to the models is a good idea.

"Fat models, skinny controllers"

File: app/controllers/admin/cafe_works_controller.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    if !@cwork.clear_worker
+      render action: show, notice: "Lyckades inte"
+    end
+  end
+
+  def setup
+    @cwork = CafeWork.new
+  end
+
+  def setup_preview
```
Comment (henrikssn) [inline app/controllers/admin/cafe_works_controller.rb]: This means the method is too big and should be split. I do agree here, maybe you could make a helper method for setup_preview and setup_create? They are very similar.

**Comment by Zofz** [comment]
Yeah, I am going to rewrite them completely. Probably moving it to the model as well.

**Comment by Zofz** [comment]
@fsek/admins 
Still needs to write proper tests.

How do we write the css-variables?

**Comment by Zofz** [comment]
@henrikssn https://travis-ci.org/fsek/web/builds/54759318
Know what is wrong here? After updating to 4.2.0

**Comment by Zofz** [comment]
@henrikssn The integration test you did with home page failed.

Otherwise I think everything is ready to go live. I'll put it up on dev first perhaps.

File: app/controllers/admin/cafe_works_controller.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  end
+
+  def main
+    @faqs = Faq.category(:Hilbert).answered
+    @faq_unanswered = Faq.category(:Hilbert).where(answer: '').count
+    @cworks = CafeWork.all
+    @cwork_grid = initialize_grid(@cworks)
+  end
+
+  private
```
Comment (Zofz) [inline app/controllers/admin/cafe_works_controller.rb]: There is a blank line?!

**Comment by Zofz** [comment]
@fsek/admins 
I do not know how to solve the CSS-things, we need to agree on how to do that!

The ```devise_parameter_sanitizer``` was too long, and now it complains on the blocks, ideas?

Otherwise ready to launch.

**Comment by henrikssn** [comment]
How to manage colors: http://sachagreif.com/sass-color-variables

I fixed cafe_works so you see how it applies to us.

File: app/controllers/contacts_controller.rb [hunk 3]
```diff
@@ -23,18 +23,19 @@ def new
   # GET /councils/1/edit
   def edit
   end
-  def mail    
-    if(params[:name]) && (params[:email]) && (params[:msg])
+
+  def mail
```
Comment (Zofz) [inline app/controllers/contacts_controller.rb]: Will have to fix when refactoring that controller.

**Comment by Zofz** [comment]
@henrikssn Kan du kika på servern? Den gillade inte min merge...

**Pull request merged**